data/mods/chatbats/moves.ts

```diff
@@ -0,0 +1,904 @@
+		shortDesc: "Encore + Rain Dance",
+	},
+	// prio + double power if opponent is using a water move
+	bonsaibounce: {
+		num: -106,
+		accuracy: 100,
+		basePower: 70,
+		category: "Physical",
+		name: "Bonsai Bounce",
+		pp: 10,
+		priority: 0,
+		flags: { protect: 1, contact: 1, mirror: 1, metronome: 1 },
+		// checks for water move usage from opponent
+		onModifyPriority(priority, source) {
+			// gets current foe in singles
+			const foe = source.side.foe.active[0];
+			if (!foe || foe.fainted) {
+				return priority;
+			}
+			// gets attack of foe this turn
+			const action = this.queue.willMove(foe);
+			if (!action || action.choice !== 'move') {
+				return priority;
+			}
+			const move = action.move;
+			if (move?.type === 'Water') {
+				return priority + 1;
+			} else {
+				return priority;
+			}
+			return priority;
+		},
+		// modifies base power
+		onBasePower(basePower, source, target) {
+			const foe = source.side.foe.active[0];
+			if (!foe || foe.fainted) {
+				return basePower;
+			}
+			const action = this.queue.willMove(foe);
+			if (!action || action.choice !== 'move') {
+				return basePower;
+			}
+			const move = action.move;
+			if (move?.type === 'Water') {
+				this.add('-message', `Sudowoodo draws power from the water!`);
+				return basePower + 70;
+			} else {
+				return basePower;
+			}
+			return basePower;
+		},
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source, move) {
+			this.add('-anim', source, 'Ivy Cudgel Rock', target);
+			this.add('-anim', source, 'Splash');
+		},
+		secondary: null,
+		target: "normal",
+		type: "Rock",
+		contestType: "Beautiful",
+		shortDesc: "+1 Priority and 2x power if target is using a Water move.",
+		desc: "If the target is using a Water type move, this move will always move first and gains double power.",
+	},
+	ironstrike: {
+		// implemented via changes to Stealth Rocks and Spikes
+		num: -107,
+		accuracy: 100,
+		basePower: 50,
+		category: "Physical",
+		name: "Iron Strike",
+		pp: 15,
+		priority: 0,
+		flags: { protect: 1, contact: 1, mirror: 1, metronome: 1 },
+		onTryMove() {
+			this.attrLastMove('[still]');
+		},
+		onPrepareHit(target, source, move) {
+			this.add('-anim', source, 'Metal Claw', target);
+		},
+		secondary: null,
+		target: "normal",
+		type: "Steel",
+		contestType: "Beautiful",
+		shortDesc: "Inflicts damage from hazards on target's side.",
+		desc: "Target takes damage from all entry hazards on their side of the field, unless they are immune.",
+	},
+	thunderouskick: {
+		inherit: true,
+		secondary: null,
+		onHit(target, source, move) {
+			// random # 0 or 1
+			const randomNum = Math.round(Math.random());
+			// 50% chance to drop def
+			if (randomNum === 0) {
+				if (target.boosts.def !== -6) {
+					this.boost({ def: -1 }, target, source, move);
+				}
+			} else {
+				this.add('-message', `${source.name} follows up with a Thunder Kick!`);
+				// defines Thunder Kick
+				const thunderKick = {
```

Comment:
it might be better to make this a full move in the file

Reply:
i dont see what mew has to do with this, and it doesnt matter if nothing actually uses it, it's just easier for readability